ocsmesh/driver.py

```diff
@@ -1,128 +1,151 @@
+            The size function definition. Must be instance of BaseHfun or None.
+        init_mesh: Mesh or MeshData, optional
+            Initial mesh or points to be used as a seed.
+        crs: str, CRS, optional
+            Coordinate reference system for the output.
+            If None, defaults to EPSG:4326.
+        engine_name : str
+            Name of the engine ('jigsaw', 'triangle', 'gmsh').
+        **engine_kwargs : dict
+            Options to pass to the engine's Option class.
+            Common options:
+            - bnd_representation (str): 'exact', 'fixed', or 'adapt'.
         """
+
         self._geom = geom
         self._hfun = hfun
-        self._init = initial_mesh
-        self._crs = CRS.from_user_input(crs) if crs is not None else crs
-        self._opts.verbosity = verbosity
-
-    def run(self, sieve=None, quality_metric=1.05, remesh_tiny_elements=False):
 
-        hfun_msh_t = self.hfun.msh_t()
+        # Set default CRS immediately if not provided
+        self._crs = CRS.from_user_input(crs) if crs is not None else CRS.from_epsg(4326)
 
-        output_mesh = jigsaw_msh_t()
-        output_mesh.mshID = 'euclidean-mesh'
-        output_mesh.ndims = 2
+        # Validate hfun
+        if hfun is not None and not isinstance(hfun, BaseHfun):
+            raise TypeError(f"hfun must be BaseHfun, got {type(hfun).__name__}")
 
-        self.opts.hfun_hmin = np.min(hfun_msh_t.value)
-        self.opts.hfun_hmax = np.max(hfun_msh_t.value)
-        self.opts.mesh_rad2 = float(quality_metric)
+        # Validate init_mesh
+        self._seed_data = None
+        if init_mesh is not None:
+            if isinstance(init_mesh, MeshData):
+                self._seed_data = init_mesh
+            elif hasattr(init_mesh, 'meshdata'):
+                self._seed_data = init_mesh.meshdata
+            else:
+                raise TypeError("init_mesh must be MeshData or have 'meshdata' prop")
 
-        geom_msh_t = self.geom.msh_t()
+        self._engine = get_mesh_engine(engine_name, **engine_kwargs)
 
-        # When the center of geom and hfun are NOT the same, utm
-        # zones would be different for resulting msh_t.
-        if geom_msh_t.crs != hfun_msh_t.crs:
-            utils.reproject(hfun_msh_t, geom_msh_t.crs)
-        output_mesh.crs = hfun_msh_t.crs
 
-        _logger.info('Calling libsaw.jigsaw() ...')
-        libsaw.jigsaw(
-            self.opts,
-            geom_msh_t,
-            output_mesh,
-            init=hfun_msh_t if self._init is True else None,
-            hfun=hfun_msh_t
+    def run(self, sieve=None) -> Mesh:
+        """
+        Run a mesh generation job.
+        """
+        shape = self._geom.geoseries()
+        calc_crs = shape.crs
+
+        # 1. Project to Metric (UTM) if needed
+        # We need metric units for 'adapt' (Hfun is in meters) and for Gmsh
+        # to correctly interpret Hfun values vs Edge lengths.
+        if calc_crs is not None and calc_crs.is_geographic:
```

Comment:
What if `calc_crs` is None? Does that mean we assume it's UTM? Since we're saying the adapt logic needs CRS lengths to be in meters